src/Client/PodcastEpisodes.php

```diff
@@ -89,13 +81,9 @@ public function slice(int $offset, int $length = null) : Sequence
                 $episodes = [];
 
                 foreach ($result['items'] as $episode) {
-                    if (isset($this->episodes[$episode['number']])) {
-                        $episodes[] = $this->episodes[$episode['number']]->wait();
-                    } else {
-                        $episodes[] = $episode = $this->denormalizer->denormalize($episode, PodcastEpisode::class,
-                            null, ['snippet' => true]);
-                        $this->episodes[$episode->getNumber()] = promise_for($episode);
-                    }
+                    return array_map(function (array $episode) {
```

Comment:
these 3 lines can probably be extracted in an helper to protect from future changes? Probably the whole callback could be 
```
->then(Denormalize::toSnippetOf(PodcastEpisode::class, $this->denormalizer))
```
```